main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -1347,6 +1411,15 @@ object ParsedAst {
     */
   case class MatchRule(pat: ParsedAst.Pattern, guard: Option[ParsedAst.Expression], exp: ParsedAst.Expression) extends ParsedAst
 
+  /**
+    * A select rule consists of a pattern, a channel expression, and a body expression.
+    *
+    * @param pat     the pattern of the rule.
+    * @param channel the channel expression of the rule.
+    * @param exp     the body expression of the rule.
+    */
+  case class SelectRule(pat: ParsedAst.Pattern, channel: ParsedAst.Expression, exp: ParsedAst.Expression) extends ParsedAst
```

Comment:
Using a pattern here is interesting. On the one hand:

```
let c : Channel[(Int, Int)] = ...
select {
  case (x, y) <- c => ...
}
```

is cool, but what about:

```
let c : Channel[Option[Int]] = ...
select {
  case Some(x) <- c => ...
}
```

what happens if c contains the value `None?` It seem there are (at least) two options for what could happen.

Reply:
Can we use a subset of Patterns.Simple in the Parser to disallow Literal and things like that? Maybe even just Patterns.variable.

Reply:
You can use Names.Variable (I think it is called).